File: frontend/src/api/conversationStarters.ts
import { api } from './client';

export interface ConversationStarter {
  prompt: string;
  context: string | null;
}

export interface ConversationStartersResponse {
  starters: ConversationStarter[];
  student_name: string;
  generated_at: string;
}

export const conversationStartersApi = {
  generate: async (studentId: number, courseId?: number | null) => {
    const response = await api.post('/api/conversation-starters/generate', {
      student_id: studentId,
      course_id: courseId ?? null,
    });
    return response.data as ConversationStartersResponse;
  },

  getDaily: async () => {
    const response = await api.get('/api/conversation-starters/daily');
    return response.data as ConversationStartersResponse;
  },
};

File: app/schemas/conversation_starters.py
"""Schemas for the Conversation Starters feature."""

from pydantic import BaseModel


class ConversationStarterRequest(BaseModel):
    student_id: int
    course_id: int | None = None


class ConversationStarter(BaseModel):
    prompt: str
    context: str | None = None


class ConversationStartersResponse(BaseModel):
    starters: list[ConversationStarter] = []
    student_name: str
    generated_at: str
